testing/unittests/unittests.cpp

```diff
@@ -928,6 +928,145 @@ CPPUNIT_TEST_SUITE_REGISTRATION( PipeRunTest );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( PipeRunTest, "PipeRunTest" );
 #endif
 
+#ifndef _WIN32
+//#define TIME_CRITSECS
+#define CHECK_CS_CONTENTION
+
+class InstrumentedCriticalSection 
+{
+private:
+    MutexId mutex;
+#ifdef _ASSERT_LOCK_SUPPORT
+    ThreadId owner = 0;
+#endif
+    unsigned depth = 0;
```

Comment:
Slightly better if this is atomic - then the compiler can't magically move the code (and use fastInc(), fastDec()).  Still get very inconsistent results though.